src/node_file.cc

```diff
@@ -2547,30 +2547,40 @@ static void ReadFileUtf8(const FunctionCallbackInfo<Value>& args) {
 
   CHECK_GE(args.Length(), 2);
 
-  BufferValue path(env->isolate(), args[0]);
-  CHECK_NOT_NULL(*path);
-
   CHECK(args[1]->IsInt32());
   const int flags = args[1].As<Int32>()->Value();
 
-  if (CheckOpenPermissions(env, path, flags).IsNothing()) return;
-
+  uv_file file;
   uv_fs_t req;
   auto defer_req_cleanup = OnScopeLeave([&req]() { uv_fs_req_cleanup(&req); });
```

Comment:
```suggestion
```

That's happening because you forgot to remove this `OnScopeLeave` call which is called (after your change) before the check for permissions, so when it throws invalid permissions, in this case /proc/cpuinfo isn't allowed, the process exits with no status code.

Removing this line might do the trick.

Reply:
Nice debugging. Thank you. Updated the pull request. Appreciate it, if you can review it once more.